armi/bookkeeping/tests/test_historyTracker.py

```diff
@@ -72,12 +72,14 @@ def setUpClass(cls):
         runTutorialNotebook()
 
         reloadCs = settings.Settings(f"{CASE_TITLE}.yaml")
+        reloadCs.lock = False
         reloadCs.caseTitle = "armiRun"
         reloadCs["db"] = True
         reloadCs["reloadDBName"] = pathlib.Path(f"{CASE_TITLE}.h5").absolute()
         reloadCs["runType"] = "Snapshots"
         reloadCs["loadStyle"] = "fromDB"
         reloadCs["detailAssemLocationsBOL"] = ["001-001"]
```

Comment:
perhaps this is a bit idealistic and more reasonable in a world where a compiler would eliminate the runtime cost of doing so, but another option to avoid an unlock here could be to express the pattern of "these settings except for [something different]" functionally, like:

```
reloadCs = settings.Settings("something.yaml")
newSettings = reloadCs.withModifications({some: dict})
```

As alluded to above, the downside is the cost of copying the settings from reloadCs to make a brand new Settings object with just a handful of different settings. But at least we would be able to walk away knowing that reloadCs (and by extension anything that holds a reference to it) is not being mutated

Reply:
Sorry, momentary paranoia here; that's out of scope for this ticket, right? This ticket was just to add a `DeprecationWarning`?

Okay, but, I like that idea. I think we also need to add some form of:

    reloadCs.save(filePath)

Because we are currently doing a `yamlize.dump()` to write YAML files in 12 different modules. And at some point we will want to add tooling to our YAML outputs, and that'll be easier to do in one place than 12.

Reply:
I'd say this is about as in/out of scope as the lock()/unlock() methods. If we just want to do a deprecation warning (assuming it doesn't produce such a firehose of horror that it's not useful), then we can just implement that and not try to make any progress on the other side to fix it. Once we mark settings modification as bad behavior, the next steps are:
 - Decide when it _is_ okay to change settings (e.g. "Any time, as long as you are honest and `unlock()` them?", "Only for some period before we `lock()` them, after which they're locked forever?", or "Never; if you want to change settings, make a new Settings object?")
 - Refactor code that is currently modifying settings to play nice with the new approach/restrictions.

To not anticipate where we will end up with the above feels a little like that `numpy` deprecation warning discussed in #246, where they say effectively "This is going to go away, but we aren't telling you what it will be replaced with." So i think it's at least worth spitballing. Perhaps this is better discussion to have in the issue (#371)...

>doing a yamlize.dump() to write YAML files in 12 different modules

Do settings use yamlize, or just Blueprints?

> `reloadCs.save(filePath)`

See https://github.com/terrapower/armi/blob/0289b7573b2ac39c792b41e1ebfe300dfd918f7b/armi/settings/caseSettings.py#L272